### PR TITLE
Rust update

### PR DESCRIPTION
**Pull request by chris-morgan**
I’ve followed your conventions. Feel free to cherrypick or modify or whatever as desired.

There is one functional difference: I unhid the `#![feature(plugin)]` lines in the documentation, because I think they are valuable.

### CODE REVIEW

File: phf/Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "phf"
 authors = ["Steven Fackler <sfackler@gmail.com>"]
-version = "0.4.6"
+version = "0.4.7"
```
Comment (sfackler) [inline phf/Cargo.toml]: Could you revert the changes to the Cargo.tomls? I like to have the version bump in a commit on its own

Comment (chris-morgan) [inline phf/Cargo.toml]: They are.

Comment (sfackler) [inline phf/Cargo.toml]: Oh! I didn't check the commits :)
